aws/data_source_aws_iam_group_test.go

```diff
@@ -29,6 +29,33 @@ func TestAccAWSDataSourceIAMGroup_basic(t *testing.T) {
 	})
 }
 
+func TestAccAWSDataSourceIAMGroup_member(t *testing.T) {
+	groupName := fmt.Sprintf("test-datasource-group-%d", acctest.RandInt())
+	userName := fmt.Sprintf("test-datasource-user-%d", acctest.RandInt())
+	groupMemberShipName := fmt.Sprintf("test-datasource-group-membership-%d", acctest.RandInt())
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsIAMGroupConfigMember(groupName, userName, groupMemberShipName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.aws_iam_group.test", "group_id"),
+					resource.TestCheckResourceAttr("data.aws_iam_group.test", "path", "/"),
+					resource.TestCheckResourceAttr("data.aws_iam_group.test", "group_name", groupName),
+					resource.TestMatchResourceAttr("data.aws_iam_group.test", "arn", regexp.MustCompile("^arn:aws:iam::[0-9]{12}:group/"+groupName)),
+					resource.TestCheckResourceAttr("data.aws_iam_group.test", "members.#", "1"),
+					resource.TestCheckResourceAttrPair("data.aws_iam_group.test", "members.0.arn", "aws_iam_user.user", "arn"),
+					resource.TestCheckResourceAttrSet("data.aws_iam_group.test", "members.0.user_id"),
```

Comment:
I believe this can be compared using the [`aws_iam_user` resource `unique_id` attribute](https://www.terraform.io/docs/providers/aws/r/iam_user.html#unique_id): 

```suggestion
					resource.TestCheckResourceAttrPair("data.aws_iam_group.test", "members.0.user_id", "aws_iam_user.user", "unique_id"),
```

Reply:
fixed